core/storage/translation/gae_models.py

```diff
@@ -0,0 +1,141 @@
+
+        Args:
+            source_language_code: str. The language code of the untranslated
+                text.
+            target_language_code: str. The language code of the translation.
+            origin_text: str. The untranslated text.
+            translated_text: str. The translation.
+
+        Returns:
+            MachineTranslatedTextModel. The newly created
+            MachineTranslatedTextModel instance.
+        """
+
+        entity_id = cls._generate_id(
+            source_language_code, target_language_code, origin_text)
+        translation_entity = cls(
+            id=entity_id,
+            source_language_code=source_language_code,
+            target_language_code=target_language_code,
+            origin_text=origin_text,
+            translated_text=translated_text)
+        translation_entity.put()
+        return entity_id
+
+    @classmethod
+    def _generate_id(
+            cls, source_language_code, target_language_code, origin_text):
+        """Generates a key for the instance of MachineTranslatedTextModel
+        class in the required format with the arguments provided.
+
+        Args:
+            source_language_code: str. The language code of the untranslated
+                text.
+            target_language_code: str. The language code of the translation.
+            origin_text: str. The untranslated text.
+
+        Returns:
+            str. The generated ID for this entity of the form
+            '[source_language_code]:[target_language_code]:[hashed_text]'.
+        """
+        # max_length is 450, less than maximum key length (500) to allow room
+        # for language_codes.
+        hashed_text = utils.convert_to_hash(origin_text, 50)
+        return (
+            '%s:%s:%s' % (
+                source_language_code, target_language_code, hashed_text)
+        )
+
+    @classmethod
+    def get_translation_for_text(
+            cls, source_language_code, target_language_code, origin_text):
+        """Gets MachineTranslatedTextModel by language codes and origin text.
+        Returns None if no translation exists for the given parameters.
+
+        Args:
+            source_language_code: str. The language of the origin_text.
+            target_language_code: str. The language being translated to.
+            origin_text: str. The text to be translated.
+
+        Returns:
+            MachineTranslatedTextModel|None. The MachineTranslatedTextModel
+            if a translation exists or None if no translation is found.
+        """
+        instance_id = cls._generate_id(
+            source_language_code, target_language_code, origin_text)
+        return cls.get(instance_id)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model does not contain user data."""
```

Comment:
Nit: Use "doesn't" to be consistent with above and below.

Reply:
Done. Is it better to prefer does not to avoid the abbreviation?